SectionedCollectionView/Views/SectionedCollectionView.swift

```diff
@@ -0,0 +1,251 @@
+    var dataSource: RxCollectionViewSectionedReloadDataSource<SectionOfCustomData>!
+    
+    let disposeBag = DisposeBag()
+    
+    var selectedLimit: Int?
+    
+    // MARK: - Outputs
+    
+    var selectedSections: Observable<[CustomData]> {
+        return sections.asObservable().map({ sections -> [CustomData] in
+            return sections.compactMap({ sections -> [CustomData] in
+                return sections.selectedItems().items
+            }).flatMap({ $0 })
+        })
+    }
+    
+    var limitReached: Observable<Void> {
+        return _limitReached.asObservable()
+    }
+    
+    // MARK: - Inputs
+    
+    var setSections: AnyObserver<[SectionOfCustomData]> {
+        return _sections.asObserver()
+    }
+    
+    // MARK: - RxSwift varable
+    private let sections = Variable<[SectionOfCustomData]>([])
+    private let _sections = PublishSubject<[SectionOfCustomData]>()
+    private let _limitReached = PublishSubject<Void>()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+       
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        
+        self.setupView()
+        self.addCollectionView()
+        self.setupCollectionView()
+        self.registerHeaderCell()
+        self.registerFooterCell()
+        self.registerCollectionViewCell()
+        self.setupDataSource()
+        self.setupBindings()
+    }
+    
+    fileprivate func setupView(){
+        backgroundColor = background
+    }
+    
+    fileprivate func addCollectionView() {
+        collectionView = UICollectionView(frame: self.frame, collectionViewLayout: UICollectionViewFlowLayout())
+        self.addSubview(collectionView)
+        collectionView.backgroundColor = .clear
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        
+        var constraints: [NSLayoutConstraint] = []
+        constraints.append(NSLayoutConstraint(item: self.collectionView, attribute: NSLayoutAttribute.top, relatedBy: NSLayoutRelation.equal, toItem: self, attribute: NSLayoutAttribute.top, multiplier: 1, constant: 0))
+        constraints.append(NSLayoutConstraint(item: self.collectionView, attribute: NSLayoutAttribute.bottom, relatedBy: NSLayoutRelation.equal, toItem: self, attribute: NSLayoutAttribute.bottom, multiplier: 1, constant: 0))
+        constraints.append(NSLayoutConstraint(item: self.collectionView, attribute: NSLayoutAttribute.leading, relatedBy: NSLayoutRelation.equal, toItem: self, attribute: NSLayoutAttribute.leading, multiplier: 1, constant: 0))
+        constraints.append(NSLayoutConstraint(item: self.collectionView, attribute: NSLayoutAttribute.trailing, relatedBy: NSLayoutRelation.equal, toItem: self, attribute: NSLayoutAttribute.trailing, multiplier: 1, constant: 0))
+        self.addConstraints(constraints)
+    }
+    
+    fileprivate func setupCollectionView() {
```

Comment:
If you use one major setupCollectionView method for all the previous and this one, this should be renamed to setupCollectionViewLayout maybe?